### PR TITLE
feat(api): add postal-code and city synchronisation

### PR DESCRIPTION
**Pull request by kwiss**

### CODE REVIEW

File: packages/api/__test__/e2e/inscription/POST_inscription.test.js [hunk 1]
```diff
@@ -125,13 +125,13 @@ test("should NOT register when pass1!==pass2", async () => {
...
     { stack: expect.any(String) },
     `
-    Object {
-      "message": "Les mots de passe ne sont pas conformes",
-      "name": "UnprocessableEntityError",
-      "stack": Any<String>,
-      "status": 422,
-    }
-  `
+            Object {
```
Comment (douglasduteil) [inline packages/api/__test__/e2e/inscription/POST_inscription.test.js]: WAT ?

File: packages/api/__test__/e2e/mandataires/__snapshots__/GET_postcode.test.js.snap [hunk 2]
```diff
@@ -2,6 +2,8 @@
 
 exports[`should get mandataires for postcode 62000 1`] = `
 Object {
+  "cities": null,
+  "code_insee": null,
```
Comment (douglasduteil) [inline packages/api/__test__/e2e/mandataires/__snapshots__/GET_postcode.test.js.snap]: Normal ?

File: packages/api/__test__/e2e/mandataires/__snapshots__/GET_postcode.test.js.snap [hunk 3]
```diff
@@ -2,6 +2,8 @@
 
 exports[`should get mandataires for postcode 62000 1`] = `
 Object {
+  "cities": null,
```
Comment (douglasduteil) [inline packages/api/__test__/e2e/mandataires/__snapshots__/GET_postcode.test.js.snap]: Normal ?

File: packages/api/routes/mandataireMesures.js [hunk 4]
```diff
@@ -373,7 +374,34 @@ router.get(
         throw createError.Unauthorized(`Mandataire not found`);
       }
       const mesures = await getMesuresMap(mandataire.id);
-      res.status(200).json(mesures);
+      const mesuresWithCityCoordinates = mesures.map(async mesure => {
+        let cityCoordinates = await getCityCoordinates(
+          mesure.ville.toUpperCase(),
+          mesure.code_postal
+        );
+        if (cityCoordinates.length > 1) {
```
Comment (douglasduteil) [inline packages/api/routes/mandataireMesures.js]: Is it a necessary if ? 
Can you :

```js
const [firstCityCoordinates] = await getCityCoordinates(...).filter(...) || [{}];

// [...]

return {
  // [...]
  latitude: firstCityCoordinates.latitude || mesure.latitude,
  longitude: firstCityCoordinates.longitude || mesure.longitude,
  // [...]
}
```

File: packages/knex/knexfile.js [hunk 5]
```diff
@@ -4,7 +4,7 @@ const { join } = require("path");
 
 module.exports = {
   test: {
-    // debug: true,
+    debug: true,
```
Comment (douglasduteil) [inline packages/knex/knexfile.js]: Can you remove re-comment the debug ?

File: packages/knex/migrations/20190607133712_add_cities_along_zip_code.js [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+  return knex.schema.alterTable("geolocalisation_code_postal", function(table) {
+    table.dropUnique("code_postal", "codepostallatlngs_code_postal_unique");
+    table.specificType("cities", "text[]");
+    table.string("code_insee");
+  });
+};
+
+exports.down = function(knex) {
+  return knex.schema.alterTable("geolocalisation_code_postal", function(table) {
+    table.dropColumn("code_postal");
```
Comment (douglasduteil) [inline packages/knex/migrations/20190607133712_add_cities_along_zip_code.js]: You should reverse the `dropUnique` here no ?

File: packages/knex/seeds/development/0050-codePostalLatLngs.js [hunk 7]
```diff
@@ -1,27 +1,34 @@
...
   return knex.transaction(tr => {
     return knex
       .table("geolocalisation_code_postal")
       .del()
       .then(() => {
         return knex
-          .batchInsert("geolocalisation_code_postal", rows)
+          .batchInsert(
+            "geolocalisation_code_postal",
+            removeUnique(newRows, "code_postal")
```
Comment (douglasduteil) [inline packages/knex/seeds/development/0050-codePostalLatLngs.js]: You have duplicates in the `zipCodes` array ?

Comment (kwiss) [reply packages/knex/seeds/development/0050-codePostalLatLngs.js]: yes

File: packages/knex/migrations/20190607133712_add_cities_along_zip_code.js (hunk 6 above)
Comment (kwiss) [reply packages/knex/migrations/20190607133712_add_cities_along_zip_code.js]: not anymore, it should be unique now, so i remove that rule

**Comment by kwiss** [comment]
> Ok for merge. But we should open some technical debt issues

=> https://github.com/SocialGouv/emjpm/issues/440

**Pull request merged**